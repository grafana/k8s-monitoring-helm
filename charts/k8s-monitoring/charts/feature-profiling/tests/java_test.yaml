# yamllint disable rule:document-start rule:line-length rule:trailing-spaces
suite: Test Java profiling
templates:
  - configmap.yaml
tests:
  - it: should build the Java profiling configuration
    set:
      deployAsConfigMap: true
      ebpf:
        enabled: false
      java:
        enabled: true
      pprof:
        enabled: false
    asserts:
      - isKind:
          of: ConfigMap
      - matchSnapshot:
          path: data["module.alloy"]
  - it: should be able to filter by label and annotation
    set:
      deployAsConfigMap: true
      ebpf:
        enabled: false
      java:
        enabled: true
        labelSelectors:
          app.kubernetes.io/name: secret-program
          colors: [blue, green]
        annotationSelectors:
          k8s.grafana.com/profile: "true"
      pprof:
        enabled: false
    asserts:
      - isKind:
          of: ConfigMap
      - matchSnapshot:
          path: data["module.alloy"]
