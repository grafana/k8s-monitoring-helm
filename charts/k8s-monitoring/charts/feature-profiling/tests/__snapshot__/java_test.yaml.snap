should be able to filter by label and annotation:
  1: |
    |-
      declare "profiling" {
        argument "profiles_destinations" {
          comment = "Must be a list of profile destinations where collected profiles should be forwarded to"
        }
        // Profiles: Java
        discovery.kubernetes "java_pods" {
          role = "pod"
          selectors {
            role = "pod"
            label = "app.kubernetes.io/name=secret-program,colors in (blue,green)"
            field = "spec.nodeName=" + sys.env("HOSTNAME")
          }
        }

        discovery.process "java_pods" {
          join = discovery.kubernetes.java_pods.targets
        }

        discovery.relabel "java_pods" {
          targets = discovery.process.java_pods.targets
          rule {
            source_labels = ["__meta_kubernetes_pod_phase"]
            regex = "Succeeded|Failed|Completed"
            action = "drop"
          }
          rule {
            source_labels = ["__meta_kubernetes_pod_name"]
            regex = "^$"
            action = "drop"
          }
          rule {
            source_labels = ["__meta_process_exe"]
            action = "keep"
            regex = ".*/java$"
          }
          rule {
            source_labels = ["__meta_kubernetes_namespace"]
            target_label = "namespace"
          }
          rule {
            source_labels = ["__meta_kubernetes_pod_annotation_k8s_grafana_com_profile"]
            regex = "true"
            action = "keep"
          }
          rule {
            source_labels = ["__meta_kubernetes_pod_name"]
            target_label = "pod"
          }
          rule {
            source_labels = ["__meta_kubernetes_pod_node_name"]
            target_label = "node"
          }
          rule {
            source_labels = ["__meta_kubernetes_pod_container_name"]
            target_label = "container"
          }
        }

        pyroscope.java "java_pods" {
          targets = discovery.relabel.java_pods.output
          profiling_config {
            interval = "60s"
            alloc = "512k"
            cpu = true
            sample_rate = 100
            lock = "10ms"
          }
          forward_to = argument.profiles_destinations.value
        }
      }
should build the Java profiling configuration:
  1: |
    |-
      declare "profiling" {
        argument "profiles_destinations" {
          comment = "Must be a list of profile destinations where collected profiles should be forwarded to"
        }
        // Profiles: Java
        discovery.kubernetes "java_pods" {
          role = "pod"
          selectors {
            role = "pod"
            field = "spec.nodeName=" + sys.env("HOSTNAME")
          }
        }

        discovery.process "java_pods" {
          join = discovery.kubernetes.java_pods.targets
        }

        discovery.relabel "java_pods" {
          targets = discovery.process.java_pods.targets
          rule {
            source_labels = ["__meta_kubernetes_pod_phase"]
            regex = "Succeeded|Failed|Completed"
            action = "drop"
          }
          rule {
            source_labels = ["__meta_kubernetes_pod_name"]
            regex = "^$"
            action = "drop"
          }
          rule {
            source_labels = ["__meta_process_exe"]
            action = "keep"
            regex = ".*/java$"
          }
          rule {
            source_labels = ["__meta_kubernetes_namespace"]
            target_label = "namespace"
          }
          rule {
            source_labels = ["__meta_kubernetes_pod_name"]
            target_label = "pod"
          }
          rule {
            source_labels = ["__meta_kubernetes_pod_node_name"]
            target_label = "node"
          }
          rule {
            source_labels = ["__meta_kubernetes_pod_container_name"]
            target_label = "container"
          }
        }

        pyroscope.java "java_pods" {
          targets = discovery.relabel.java_pods.output
          profiling_config {
            interval = "60s"
            alloc = "512k"
            cpu = true
            sample_rate = 100
            lock = "10ms"
          }
          forward_to = argument.profiles_destinations.value
        }
      }
